tests/test_views.py

```diff
@@ -519,6 +523,95 @@ async def test_notifications_with_disabled_option(
     )
     assert resp.status == HTTPStatus.FORBIDDEN
 
+async def test_notifications_with_no_expiration(
+    hass_client_local_frigate: Any,
+    hass: Any,
+) -> None:
+    """Test that notification events are served if they are set to not expire."""
+
+    # Make another config entry with the same data but with
+    # CONF_NOTIFICATION_PROXY_ENABLE disabled.
+    private_config_entry = create_mock_frigate_config_entry(
+        hass,
+        entry_id="private_id",
+        options={CONF_EXPIRE_NOTIFICATIONS_AFTER_MINS: 0},
+        data=hass.config_entries.async_get_entry(TEST_CONFIG_ENTRY_ID).data,
+    )
+
+    private_config: dict[str, Any] = copy.deepcopy(TEST_CONFIG)
+    private_config[ATTR_MQTT][ATTR_CLIENT_ID] = "private_id"
+    private_client = create_mock_frigate_client()
+    private_client.async_get_config = AsyncMock(return_value=private_config)
+
+    await setup_mock_frigate_config_entry(
+        hass, config_entry=private_config_entry, client=private_client
+    )
+
+    # Fake time is 2021-11-01T19:02:00
+    with patch("custom_components.frigate.views.NotificationsProxyView._get_current_datetime",
```

Comment:
Why not just patch datetime.datetime.now directly or equivalent? You may be able to copy what I  did here and not need the static method above.

https://github.com/blakeblackshear/frigate-hass-integration/blob/master/tests/test_media_source.py#L44

Reply:
This is pretty neat, thank you! I was not able to simply do `with patch("custom_components.frigate.views.datetime.datetime.now` because it would complain about patching a built-in function.